### PR TITLE
dx12: Implement buffer views

### PR DESCRIPTION
**Pull request by msiglreith**
Untested so far, probably requires small changes to portability due to API breakage.

### CODE REVIEW

File: src/backend/dx12/src/device.rs [hunk 1]
```diff
@@ -2193,7 +2274,16 @@ impl d::Device<B> for Device {
                         dst_samplers.push(bind_info.sampler_range.as_ref().unwrap().at(offset));
                         num_samplers.push(1);
                     }
-                    pso::Descriptor::TexelBuffer(_) => unimplemented!()
+                    pso::Descriptor::UniformTexelBuffer(buffer_view) => {
+                        src_views.push(buffer_view.handle_srv);
```
Comment (kvark) [inline src/backend/dx12/src/device.rs]: given that the SRV here and UAV below can be NULL, can we at least `error!()` on it?

**Comment by kvark** [comment]
Looks like Metal needs a few more corrections:
https://travis-ci.org/gfx-rs/gfx/jobs/374861471#L2659

**Comment by msiglreith** [comment]
bors r=kvark

**Pull request merged**